hyperspy/models/edsmodel.py

```diff
@@ -0,0 +1,881 @@
+
+        After the fit, the background is fixed.
+
+        Parameters
+        ----------
+        start_energy : {float, None}
+            If float, limit the range of energies from the left to the
+            given value.
+        end_energy : {float, None}
+            If float, limit the range of energies from the right to the
+            given value.
+        windows_sigma: list of two float
+            The upper and lower bounds around each X-ray lines to define
+            the energy range free of X-ray lines.
+        kind : {'single', 'multi'}
+            If 'single' fit only the current location. If 'multi'
+            use multifit.
+        **kwargs : extra key word arguments
+            All extra key word arguments are passed to fit or
+
+        See also
+        --------
+        free_background
+        """
+
+        if end_energy is None:
+            end_energy = self.end_energy
+        if start_energy is None:
+            start_energy = self.start_energy
+
+        # disactivate line
+        self.free_background()
+        with stash_active_state(self):
+            self.disable_xray_lines()
+            self.set_signal_range(start_energy, end_energy)
+            for component in self:
+                if component.isbackground is False:
+                    self.remove_signal_range(
+                        component.centre.value -
+                        windows_sigma[0] * component.sigma.value,
+                        component.centre.value +
+                        windows_sigma[1] * component.sigma.value)
+            if kind == 'single':
+                self.fit(**kwargs)
+            if kind == 'multi':
+                self.multifit(**kwargs)
+            self.reset_signal_range()
+        self.fix_background()
+
+    def _twin_xray_lines_width(self, xray_lines):
+        """
+        Twin the width of the peaks
+
+        The twinning models the energy resolution of the detector
+
+        Parameters
+        ----------
+        xray_lines: list of str or 'all_alpha'
+            The Xray lines. If 'all_alpha', fit all using all alpha lines
+        """
+        if xray_lines == 'all_alpha':
+            xray_lines = [compo.name for compo in self.xray_lines]
+
+        for i, xray_line in enumerate(xray_lines):
+            component = self[xray_line]
+            if i == 0:
+                component_ref = component
+                component_ref.sigma.free = True
+                E_ref = component_ref.centre.value
+            else:
+                component.sigma.free = True
+                E = component.centre.value
+                component.sigma.twin_function = _get_sigma(
+                    E, E_ref, self.units_factor)
+                component.sigma.twin_inverse_function = _get_sigma(
+                    E_ref, E, self.units_factor)
+                component.sigma.twin = component_ref.sigma
+
+    def _set_energy_resolution(self, xray_lines, *args, **kwargs):
```

Comment:
as far as I can see we don't so long as ref is added as an argument - so I've done that.